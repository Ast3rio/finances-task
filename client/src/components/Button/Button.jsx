import React from 'react';
import styles from "./Button.module.scss";

export const Button = ({label, active, onClick}) => {
    return (
        <button
            className={`${styles.btn} ${active && styles.active}`}
            onClick={onClick}
        >
            {label}
        </button>
    );
};

