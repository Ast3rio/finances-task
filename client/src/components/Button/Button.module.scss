.btn{
  width: 150px;
  height: 42px;
  color: #fff;
  background-color: #211f1f;
  outline: none;
  border: none;
  border-radius: 8px;
  cursor: pointer;
  transition: all .3s;
  &.active{
    background-color: #1e4e0e;
  }
}
