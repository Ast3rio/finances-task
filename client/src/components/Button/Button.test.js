import React from "react";
import {Button} from "./Button";
import renderer from 'react-test-renderer';

describe('Test <Button/>', () => {
    it('Button have renderer correctly', () => {
        const button = renderer.create(<Button />).toJSON();
        expect(button).toMatchSnapshot();
    })
})
