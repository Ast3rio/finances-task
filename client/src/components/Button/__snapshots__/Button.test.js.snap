// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button/> Button have renderer correctly 1`] = `
<button
  className="btn undefined"
/>
`;
